benchmarks/datasets/join/generate.sql

```diff
@@ -0,0 +1,401 @@
+    "fill17",
+    "fill18",
+    "fill19",
+    "fill20",
+    "fill21",
+    "fill22",
+    "fill23",
+    "fill24",
+    "fill25",
+    "fill26",
+    "fill27",
+    "fill28"
+)
+SELECT
+    fgs.new_file_id AS "id",
+    dil."id" AS "documentId", -- Join to get the actual document ID based on the random row number
+    'File Content Section A: ' || md5(random()::text) || E'\nFile Content Section B (metadata): ' || md5(random()::text) || E'\nAssociated ID: ' || (random()*100000)::INT AS "content",
+    CASE (random() * 10)::INT -- Introduce 'collab12' in roughly 10% of rows
+        WHEN 0 THEN 'collab12 ' || substring(md5(random()::text), 1, 20)
+        WHEN 1 THEN 'FY2025_BUDGET_DATA ' || substring(md5(random()::text), 1, 10)
+        ELSE 'File Title ' || (random()*50000)::INT || ' ' || substring(md5(random()::text), 1, 22)
+    END AS "title",
+    'File Parent Identifier: ' || substring(md5(random()::text), 1, 18) AS "parents",
+    (random()*10000)::BIGINT AS "sizeInBytes",
+    '2023-01-01 00:00:00'::TIMESTAMP + random() * ('2023-12-31 23:59:59'::TIMESTAMP - '2023-01-01 00:00:00'::TIMESTAMP) AS "createdAt",
+    (random()*20000000)::BIGINT AS "fill0",
+    (random()*20000000)::BIGINT AS "fill1",
+    (random()*20000000)::BIGINT AS "fill2",
+    (random()*20000000)::BIGINT AS "fill3",
+    (random()*20000000)::BIGINT AS "fill4",
+    (random()*20000000)::BIGINT AS "fill5",
+    (random()*20000000)::BIGINT AS "fill6",
+    (random()*20000000)::BIGINT AS "fill7",
+    (random()*20000000)::BIGINT AS "fill8",
+    (random()*20000000)::BIGINT AS "fill9",
+    md5(random()::text) "fill10",
+    md5(random()::text) "fill11",
+    md5(random()::text) "fill12",
+    md5(random()::text) "fill13",
+    md5(random()::text) "fill14",
+    md5(random()::text) "fill15",
+    md5(random()::text) "fill16",
+    md5(random()::text) "fill17",
+    md5(random()::text) "fill18",
+    md5(random()::text) "fill19",
+    md5(random()::text) "fill20",
+    md5(random()::text) "fill21",
+    md5(random()::text) "fill22",
+    md5(random()::text) "fill23",
+    md5(random()::text) "fill24",
+    md5(random()::text) "fill25",
+    md5(random()::text) "fill26",
+    md5(random()::text) "fill27",
+    md5(random()::text) "fill28"
+FROM file_generation_series fgs
+JOIN document_id_list dil ON fgs.random_doc_rn = dil.rn;
+
+
+WITH file_id_list AS (
+    -- Select all file IDs and assign a unique row number after randomizing their order
+    SELECT "id", row_number() OVER (ORDER BY random()) AS rn
+    FROM files
+),
+file_id_count AS (
+    -- Get the total count of available file IDs
+    SELECT count(*) AS total_files FROM file_id_list
+),
+page_generation_series AS (
+    SELECT
+        generate_series(1, :rows) AS series_idx,
+        gen_random_uuid()::text AS new_page_id,
+        -- This will be a random number between 1 and total_files (inclusive)
+        floor(random() * (SELECT total_files FROM file_id_count) + 1)::bigint AS random_file_rn
+)
```

Comment:
This will need to be improved to make running this at scale reasonable. Rather than joining, we should probably make the ids in documents and files effectively-monotonic (but still large strings), and then choose randomly in the range that we know is populated.

Reply:
Inlined as a `TODO`, since this dataset is already useful for smaller datasets.